tests/platform/fw_utility/test_fwutil.py

```diff
@@ -0,0 +1,501 @@
+    localhost.wait_for(host=dut.hostname, port=22, state="started", delay=10, timeout=reboot_ctrl_dict['timeout'])
+
+    # logging.info("Check the uptime to verify whether reboot was performed")
+    # dut_uptime = datetime.strptime(dut.command("uptime -s")["stdout"], "%Y-%m-%d %H:%M:%S")
+    # assert float(dut_uptime.strftime("%s")) - float(dut_datetime.strftime("%s")) > 10, "Device did not reboot"
+
+    logging.info("Wait until all critical services are fully started")
+    check_critical_services(dut)
+
+    logging.info("Check pmon daemon status")
+    assert check_pmon_daemon_status(dut), "Not all pmon daemons running."
+
+    if dut.facts["asic_type"] in ["mellanox"]:
+        pdb.set_trace()
+        current_file_dir = os.path.dirname(os.path.realpath(__file__))
+        parent_dir = os.path.abspath(os.path.join(current_file_dir, os.pardir))
+        sub_folder_dir = os.path.join(parent_dir, "mellanox")
+        if sub_folder_dir not in sys.path:
+            sys.path.append(sub_folder_dir)
+        from check_hw_mgmt_service import check_hw_management_service
+        from check_sysfs import check_sysfs
+
+        logging.info("Check the hw-management service")
+        check_hw_management_service(dut)
+
+        logging.info("Check sysfs")
+        check_sysfs(dut)
+
+
+@pytest.fixture(scope='function')
+def psu_ctrl(testbed_devices, psu_controller):
+    """
+    return psu_controller
+    """
+    pdb.set_trace()
+    dut = testbed_devices["dut"]
+    ctrl = psu_controller(dut.hostname, dut.facts["asic_type"])
+    yield ctrl
+
+
+def cpld_update(request, dut):
+    """
+    performs 30 sec power cycle off to finish cpld installation.
+    """
+    pdb.set_trace()
+    cmd_num_psu = "sudo psuutil numpsus"
+    logging.info("Check how much PSUs DUT has")
+    psu_num_out = dut.command(cmd_num_psu)
+    psu_num = 0
+    try:
+        psu_num = int(psu_num_out["stdout"])
+    except:
+        assert False, "Unable to get the number of PSUs using command '%s'" % cmd_num_psu
+
+    logging.info("Create PSU controller for testing")
+    psu_control = request.getfixturevalue("psu_ctrl")
+    if psu_control is None:
+        pytest.fail("No PSU controller for %s, skip rest of the testing in this case" % dut.hostname)
+    pdb.set_trace()
+    all_psu_status = psu_control.get_psu_status()
+    if all_psu_status:
+        for psu in all_psu_status:
+            if psu["psu_on"]:
+                psu_control.turn_off_psu(psu["psu_id"])
+                time.sleep(5)
+
+        # perform 30 seconds timeout
+        time.sleep(30)
+
+        all_psu_status = psu_control.get_psu_status()
+        if all_psu_status:
+            # turn on all psu
+            for psu in all_psu_status:
+                if not psu["psu_on"]:
+                    psu_control.turn_on_psu(psu["psu_id"])
+                    time.sleep(5)
+
+    # wait for dut to go up
+    time.sleep(20)
+
+
+def check_bios_version(version_to_install, comp_data):
```

Comment:
@noaOrMlnx for auxiliary stuff like this i would recommend to create a separate class / module.

Reply:
Done